File: atlas/examples/kotlin/application/src/main/kotlin/dev/atlas/example/application/catalog/CatalogSeedReader.kt
package dev.atlas.example.application.catalog

import dev.atlas.example.domain.catalog.CatalogItemDraft

/**
 * Loads validated seed drafts from an external representation.
 */
interface CatalogSeedReader {
    /**
     * Reads all drafts available to the current import operation.
     *
     * @return Validated catalog drafts in document order.
     */
    fun read(): List<CatalogItemDraft>
}

File: atlas/examples/typescript/packages/application/src/catalog/CatalogSeedReader.ts
import type { CatalogItemDraft } from '@atlas-example/domain';

/**
 * Loads validated seed drafts from an external representation.
 */
export interface CatalogSeedReader {
  /**
   * Reads all drafts available to the current import operation.
   *
   * @returns Validated catalog drafts in document order.
   */
  read(): readonly CatalogItemDraft[];
}
